### PR TITLE
Precious

### PR DESCRIPTION
**Pull request by Pr3c10us**

### CODE REVIEW

File: backend/app.js [hunk 1]
```diff
@@ -1,21 +1,79 @@
-const express = require('express');
-const cors = require('cors');
+// Import required modules
+require("dotenv").config();
```
Comment (ikwuoz) [inline backend/app.js:2]: use the `.env` in root dir

File: backend/controller/vm.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    DefaultAzureCredential,
+} = require("@azure/identity");
+const { ComputeManagementClient } = require("@azure/arm-compute");
+const { ResourceManagementClient } = require("@azure/arm-resources");
+const { NetworkManagementClient } = require("@azure/arm-network");
+require("dotenv").config();
+const virtualMachineServiceId = "vm";
+
+const createVm = async (req, res) => {
+    const { serviceId, userAddr, duration } = req.body;
```
Comment (ikwuoz) [inline backend/controller/vm.js:17]: is there a provision for the `signature` meant for authentication?

File: backend/controller/vm.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    } catch (error) {
+        // if (error.code === "ResourceGroupNotFound") {
+        //     res.status(404).json({ msg: "Resource not found" });
+        // } else {
+        throw new BadRequestError(error.message);
+        // }
+    }
+};
+
+const stopVm = async (req, res) => {
```
Comment (ikwuoz) [inline backend/controller/vm.js:104]: this should also be authenticated by `signature`

**Review by ikwuoz — COMMENTED** [review]
Please check the comments.

File: backend/controller/vm.js (hunk 3 above)
Comment (Pr3c10us) [reply backend/controller/vm.js:104]: so whe its time i would create a middleware that would handle the auth with signature

File: backend/app.js (hunk 1 above)
Comment (Pr3c10us) [reply backend/app.js:2]: i don't get

File: backend/controller/vm.js (hunk 2 above)
Comment (Pr3c10us) [reply backend/controller/vm.js:17]: yh

File: backend/app.js (hunk 1 above)
Comment (ikwuoz) [reply backend/app.js:2]: you can add the path to the `config` function, ideally this path should point to the `.env` in the project's root directory.

```require('dotenv').config({ path: '/custom/path/to/.env' })```

File: backend/controller/vm.js (hunk 3 above)
Comment (ikwuoz) [reply backend/controller/vm.js:104]: alright.

File: backend/app.js (hunk 1 above)
Comment (Pr3c10us) [reply backend/app.js:2]: okay

**Review by ikwuoz — COMMENTED** [review]
would merge

**Pull request merged**